llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1487,6 +1492,14 @@ private SelectHostResult selectHost(TaskInfo request) {
         return SELECT_HOST_RESULT_DELAYED_RESOURCES;
       }
 
+      // When all nodes are busy, reset locality delay
+      if (activeNodesWithFreeSlots.isEmpty()) {
+        isCapacityFull.set(true);
+        if (request.localityDelayTimeout > 0 && isRequestedHostPresent(request)) {
+          request.resetLocalityDelayInfo();
+        }
+      }
```

Comment:
isCapacityFull can be set to false in else condition here? instead of trySchedulingPendingTasks.. just a minor readability improvement..